### PR TITLE
[Substrate.io Migration] - Polkadot Protocol > Protocol Components > Transactions > Transactions Formats

### PR DESCRIPTION
**Pull request by nhussein11**
This PR aims to migrate this page of the substrate tutorials: https://docs.substrate.io/learn/transaction-formats/

Disclaimer: I did not migrate this section: https://docs.substrate.io/reference/transaction-format/#:~:text=Polkadot%20JS%20Apps%20example%3A 

Since I believe that would fit better in the tutorials section, maybe we can add a reference to that part in the future. However, I'm open to discuss about it :)

### CODE REVIEW

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (CrackTheCode016) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:3]: Missing description

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (CrackTheCode016) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:14]: Might be a good idea to add some examples of unchecked, checked, and opaque types in the codebase

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (CrackTheCode016) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:15]: I suggest we also define what 'inherent' means in a note or similar before or after this list.

```suggestion
- Checked - inherent extrinsics, which, by definition, don't require signature verification. Instead, they carry information on where the extrinsic comes from and some extra data
```

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (CrackTheCode016) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:34]: This is fine for now, but maybe a mermaid chart for stuff like this in the future 👀

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (CrackTheCode016) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:46]: We could link the necessary docs for this eventually, most likely runtime upgrades which might detail what exactly these things are

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (CrackTheCode016) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:2]: Not sure if it was ever discussed, maybe "Extrinsic Formats"

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (CrackTheCode016) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:79]: Maybe add links from PSDK Rust docs?

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (CrackTheCode016) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:71]: There is also a reference doc in the Polkadot SDK docs that would be good to link here, and also contains most important types: https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/reference_docs/signed_extensions/index.html

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (CrackTheCode016) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:92]: Links here as well

**Review by CrackTheCode016 — COMMENTED** [review]
Overall good, just a few suggestions for the content

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (nhussein11) [reply polkadot-protocol/protocol-components/transactions/transactions-formats.md:49]: Will add a TODO:update-path so when that docs is ready, we can link it up

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (nhussein11) [reply polkadot-protocol/protocol-components/transactions/transactions-formats.md:2]: I do like "Extrinsic Formats" more, but as the name of the whole section is "Transactions," I'm unsure about this one

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (nhussein11) [reply polkadot-protocol/protocol-components/transactions/transactions-formats.md:74]: Thanks!! <3

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (CrackTheCode016) [reply polkadot-protocol/protocol-components/transactions/transactions-formats.md:2]: Yeah, we'd have to rename it

Comment (nhussein11) [reply polkadot-protocol/protocol-components/transactions/transactions-formats.md:2]: @0xLucca @eshaben @dawnkelly09 do you guys agree on this?

Comment (CrackTheCode016) [reply polkadot-protocol/protocol-components/transactions/transactions-formats.md:2]: We can always do this kind of renaming later too, as to not hold up merging of PRs

Comment (nhussein11) [reply polkadot-protocol/protocol-components/transactions/transactions-formats.md:2]: Yes please, let's do that 🙏 

Anyways, will jot this down to raise this conversation in the future. Let's move on with the formatting review now 🚀

Comment (nhussein11) [reply polkadot-protocol/protocol-components/transactions/transactions-formats.md:2]: If are ok with the content, please do approve the PR so we erin and dawn can take a look at it

Comment (0xLucca) [reply polkadot-protocol/protocol-components/transactions/transactions-formats.md:2]: It would make sense to rename it to extrinsics. Lets discuss this later

**Review by CrackTheCode016 — DISMISSED** [review]
overall lgtm, the back linking to the psdk is especially important here

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (0xLucca) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:40]:
```suggestion
    - The parameters required by the function call being targeted
```

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (0xLucca) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:49]: What would you link in compatible runtime?

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (0xLucca) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:67]: This should link SCALE

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (0xLucca) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:76]: Link the SignedExtension trait

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (0xLucca) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:84]:
```suggestion
- `AdditionalSigned` - to handle any additional data to go into the signed payload. This allows you to attach any custom logic prior to dispatching a transaction
```

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (0xLucca) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:87]: Link this examples

**Comment by dawnkelly09** [comment]
Pushed a few formatting changes

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (eshaben) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:16]:
```suggestion
- [**Unchecked**](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/primitives/runtime/src/generic/unchecked_extrinsic.rs#L70){target=\_blank} - signed transactions requiring some validation check before being accepted in the transaction pool. Any unchecked extrinsic contains the signature for the sent data and some extra data
- [**Checked**](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/primitives/runtime/src/generic/checked_extrinsic.rs#L35){target=\_blank} - inherent extrinsics, which, by definition, don't require signature verification. Instead, they carry information on where the extrinsic comes from and some extra data
- [**Opaque**](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/primitives/runtime/src/lib.rs#L915){target=\_blank} - used for cases when an extrinsic hasn't yet been committed to a format but can still be decoded
```

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (eshaben) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:76]:
```suggestion
The Polkadot SDK provides the concept of [signed extensions](https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/reference_docs/signed_extensions/index.html){target=\_blank}, which extend an extrinsic with additional data, through the [`SignedExtension`](https://paritytech.github.io/try-runtime-cli/sp_runtime/traits/trait.SignedExtension.html){target=\_blank} trait.
```

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (eshaben) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:85]:
```suggestion
- [**`AccountId`**](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/primitives/runtime/src/traits.rs#L1489){target=\_blank} - to encode the sender's identity
- [**`Call`**](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/primitives/runtime/src/traits.rs#L1492){target=\_blank} - to encode the pallet call to be dispatched. This data is used to calculate transaction fees
- [**`AdditionalSigned`**](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/primitives/runtime/src/traits.rs#L1496){target=\_blank} - to handle any additional data to go into the signed payload. This allows you to attach any custom logic prior to dispatching a transaction
- [**`Pre`**](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/primitives/runtime/src/traits.rs#L1499){target=\_blank} - to encode the information that can be passed from before a call is dispatched to after it gets dispatched
```

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (eshaben) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:82]: Ideally, we don't link to specific lines. We gotta break out of this pattern because the files can change all the time and then we'll be referring people to the incorrect lines

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (eshaben) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:106]:
```suggestion
- [`txwrapper-core`](https://github.com/paritytech/txwrapper-core){target=\_blank} - for offline transactions
```

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (eshaben) [inline polkadot-protocol/protocol-components/transactions/transactions-formats.md:107]:
```suggestion
- [`@substrate/api-sidecar`](https://github.com/paritytech/substrate-api-sidecar){target=\_blank} - for REST-based transactions
```
These last two suggestions just default to using the names in the READMEs

**Review by eshaben — COMMENTED** [review]
Some minor formatting stuff. But I also left a comment about linking to specific lines. I didn't actually check to see if we could link to another part of the Rust docs, but can you please look into it and see if we can do that instead of specific file lines?

File: polkadot-protocol/protocol-components/transactions/transactions-formats.md
Comment (nhussein11) [reply polkadot-protocol/protocol-components/transactions/transactions-formats.md]: Totally agree! I didn’t notice those comments on the last PR before tackling this one, my bad! 🙏

IMO, we should set this idea as a rule of thumb for future PRs

**Comment by nhussein11** [comment]
Just requesting re-review for pending approvals

**Review by CrackTheCode016 — APPROVED** [review]

**Review by eshaben — APPROVED** [review]
LGTM!

**Comment by eshaben** [comment]
@nhussein11 just some merge conflicts to resolve

**Review by CrackTheCode016 — APPROVED** [review]

**Pull request merged**
